server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -1291,6 +1298,20 @@ public void failEngine(String reason, @Nullable Exception failure) {
                             );
                         }
                     }
+
+                    // If cleanup unreferenced flag is enabled and force merge or regular merge failed due to disk full,
+                    // cleanup all unreferenced files created during failed and reset the shard state back to last
+                    // Lucene Commit.
+                    if (shouldCleanupUnreferencedFile()
+                        && (reason.equals("force merge") || reason.equals("merge failed"))
```

Comment:
Do we have existing constants for `force merge` and `merge failed` reasons? If not, should we create these as constants?

Reply:
Addressed the comments.